File: backend/src/main/java/dev/gabus/dto/Asistencia/Asistencia.java
package dev.gabus.dto.Asistencia;

import java.time.LocalDate;

import dev.gabus.dto.Estudiante.Estudiante;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import jakarta.persistence.UniqueConstraint;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "asistencia", uniqueConstraints = {
    @UniqueConstraint(columnNames = {"estudiante_id", "fecha"})
})
public class Asistencia {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "estudiante_id", nullable = false)
    private Estudiante estudiante;

    @Column(nullable = false)
    private LocalDate fecha;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private EstadoAsistencia estado;

    private String observacion;

    public enum EstadoAsistencia {
        PRESENTE,
        AUSENTE,
        TARDE,
        EXCUSA
    }
}

File: frontend/src/app/services/asistencia.ts
import { Estudiante } from "./estudiante";

export enum EstadoAsistencia {
    PRESENTE = 'PRESENTE',
    TARDE = 'TARDE',
    AUSENTE = 'AUSENTE',
    EXCUSADO = 'EXCUSADO'
}

export interface Asistencia {
    id: number;
    estudiante: Estudiante;
    fecha: string;
    hora: string;
    estado: EstadoAsistencia;
}

export interface AsistenciaRequest {
    estudianteId?: number;
    nfcId?: string;
    fecha?: string;
    hora?: string;
    estado: EstadoAsistencia;
}
